SimG4CMS/HGCalTestBeam/plugins/HGCalTBAnalyzer.cc

```diff
@@ -560,14 +560,14 @@ void HGCalTBAnalyzer::analyzeSimHits (int type, std::vector<PCaloHit>& hits) {
 #endif
     hSimHitLng1_[type]->Fill(layer,energy);
     if (type == 0) {
-      if (layer<(int)(hSimHitLayEn2E_.size())) {
-	simHitLayEn2E[layer] = energy;
-	hSimHitLayEn2E_[layer]->Fill(energy);
+      if (layer-1 < (int)(hSimHitLayEn2E_.size())) {
```

Comment:
note that layer is never used aside from layer-1 - perhaps wise to define layer= itr->first -1; ? @bsunanda